### ISSUE TITLE
post-1.11-rebase: reenable event handling while image pruning

### ISSUE THREAD

**Issue by miminar** [openshift/origin#20127]
Underlying gonum/graph assigns conflicting IDs to new nodes after a node deletion. Thus two different nodes amy be assigned the same ID. This leads to panic when an edge is set between them.

The following PR temporarily disables event handling until the package is fixed. This ensures that no new nodes will be added to the graph after the first node gets removed.

https://github.com/deads2k/origin/pull/87

Either gonum/graph needs to be fixed or some replacement needs to be found.

The code is still buggy in 3.10 and needs to be fixed. The rebase just made the bug visible.

**Comment by bparees** [comment]
we'll also need to fix this behavior in 3.10.z

**Comment by bparees** [comment]
Does gonum/graph allow us to generate our own ids?

**Comment by miminar** [comment]
> Does gonum/graph allow us to generate our own ids?

Good idea. The generation is quite simple: https://github.com/openshift/origin/blob/master/vendor/github.com/gonum/graph/simple/directed.go#L51

We can use our own function though.

**Comment by rishi-jat** [comment]
@bparees, @miminar can I work on this issue ?

**Comment by adambkaplan** [comment]
@rishi-jat this issue predates OpenShift 4.0, I doubt it is relevant any more. It may have in fact been fixed elsewhere.
